src/gennylib/src/context.cpp

```diff
@@ -42,7 +43,7 @@ WorkloadContext::WorkloadContext(YAML::Node node,
     mongocxx::instance::current();
 
     // TODO: make this optional and default to mongodb://localhost:27017
-    auto poolFactory = PoolFactory(mongoUri);
+    auto poolFactory = PoolFactory(mongoUri, clientOpts);
```

Comment:
I think this muddies how the connection should be configured - previously it was only from the uri and/or yaml configs and PoolFactory worked out how to munge those together. Now there is an additional `clientOpts` in the works. Is there a way we can push this concern into the poolfactory so there's not another layer of indirection?